homeassistant/components/unifiprotect/__init__.py

```diff
@@ -83,49 +84,88 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    if (
+        not entry.options.get(CONF_ALLOW_EA, False)
+        and await nvr_info.get_is_prerelease()
+    ):
         ir.async_create_issue(
             hass,
             DOMAIN,
-            f"ea_warning_{protect_version}",
-            is_fixable=False,
-            is_persistent=False,
+            "ea_warning",
+            is_fixable=True,
+            is_persistent=True,
             learn_more_url="https://www.home-assistant.io/integrations/unifiprotect#about-unifi-early-access",
             severity=IssueSeverity.WARNING,
             translation_key="ea_warning",
-            translation_placeholders={"version": str(protect_version)},
+            translation_placeholders={"version": str(nvr_info.version)},
+            data={"entry_id": entry.entry_id},
         )
 
+    try:
+        await _async_setup_entry(hass, entry, data_service)
+        raise Exception("test")
+    except Exception as err:
+        if await nvr_info.get_is_prerelease():
+            # If they are running a pre-release, its quite common for setup
+            # to fail so we want to create a repair issue for them so its
+            # obvious what the problem is.
+            ir.async_create_issue(
+                hass,
+                DOMAIN,
+                f"ea_setup_failed_{nvr_info.version}",
+                is_fixable=False,
+                is_persistent=False,
+                learn_more_url="https://www.home-assistant.io/integrations/unifiprotect#about-unifi-early-access",
+                severity=IssueSeverity.ERROR,
+                translation_key="ea_setup_failed",
+                translation_placeholders={
+                    "error": str(err),
+                    "version": str(nvr_info.version),
+                },
+            )
+            ir.async_delete_issue(hass, DOMAIN, "ea_warning")
+            _LOGGER.exception("Error setting up UniFi Protect integration: %s", err)
+        raise
+
     return True
 
 
+async def _async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, data_service: ProtectData
+) -> None:
+    await async_migrate_data(hass, entry, data_service.api)
+
+    await data_service.async_setup()
+    if not data_service.last_update_success:
+        raise ConfigEntryNotReady
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = data_service
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+    async_setup_services(hass)
+    hass.http.register_view(ThumbnailProxyView(hass))
+    hass.http.register_view(VideoProxyView(hass))
+
+
 async def _async_options_updated(hass: HomeAssistant, entry: ConfigEntry) -> None:
     """Update options."""
     await hass.config_entries.async_reload(entry.entry_id)
 
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload UniFi Protect config entry."""
+
+    # entry was never fully set up (EA repair active)
+    if entry.entry_id not in hass.data[DOMAIN]:
+        return True
+
```

Comment:
I think we might be able to drop this now

Reply:
I would still need to move [this line](https://github.com/home-assistant/core/pull/81758/files#diff-18ed302c3a0144b0a47a86f87d879c8e573cb344aad2c1d793c969841c08011bR150) out of the try/except block.

Reply:
I originally added it though because if one of the entities platforms error, it causes more exceptions to try to unload them when they were not loaded. But I guess that is okay since we already told the user at this point there was errors and issues using the EA version.